pkg/provisioningv2/rke2/planner/agent.go

```diff
@@ -37,5 +40,22 @@ func (p *Planner) generateClusterAgentManifest(controlPlane *rkev1.RKEControlPla
 		return nil, err
 	}
 
-	return systemtemplate.ForCluster(mgmtCluster, tokens[0].Status.Token, taints)
+	var formattedPrivateRegistry map[string][]byte
+	privateRegistrySecret := mgmtCluster.GetSecret("PrivateRegistrySecret")
```

Comment:
Should we make `PrivateRegistrySecret` and `PrivateRegistryURL` constants? I see them used in multiple places, which introduces the opportunity for mistake.

Reply:
Yeah that's a good idea, I'll add one into `cluster_types.go`, but if there's a better place lmk